musicround/routes/core.py

```
@@ -100,13 +109,15 @@ def search_results():
                 response.raise_for_status()
                 results = response.json()
                 
-                # Check if the token was refreshed by Authlib
-                # The new token would be in oauth.spotify.token
-                if oauth.spotify.token and oauth.spotify.token.get('access_token') != authlib_token.get('access_token'):
+                # Check if the token was refreshed by Authlib (only possible when
+                # using the user's own token, since that's the only one with a
+                # refresh_token attached above).
+                if (token_source == 'user' and oauth.spotify.token
```

Ruff (0.15.20)

[error] Possible hardcoded password assigned to: "token_source"

(S105)